### ISSUE TITLE
[BUG] Tests cannot run with Node.js 17.x

### ISSUE THREAD

**Issue by targos** [npm/cli#3919]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Tests cannot run with Node.js 17.x because a test dependency ([import-jsx](https://github.com/vadimdemedes/import-jsx/issues/16)) uses an algorithm that is not supported by default in OpenSSL 3.

See https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/2785/nodes=fedora-last-latest-x64/testReport/junit/(root)/citgm/npm_v8_1_0/

```
 Error: error:0308010C:digital envelope routines::unsupported
     at new Hash (node:internal/crypto/hash:67:19)
     at Object.createHash (node:crypto:130:10)
     at filename (/home/iojs/tmp/citgm_tmp/3d350d06-273d-4e0d-b2f7-aea0cf08fa80/npm/node_modules/tap/node_modules/import-jsx/cache.js:23:22)
```

/cc @MylesBorins

**Comment by voxpelli** [comment]
Note: Since its the reporter (more precisely `import-jsx`) that's the issue, using another/simpler reporter is a workaround. Eg. this works locally on Node.js 17.1.0:

```bash
npx tap -Rtap
```

**Comment by fritzy** [comment]
This is fixed in npm@8.3.1 due to an upgrade in tap.
